src/sentry/message_filters.py

```diff
@@ -107,7 +80,7 @@ def get_filter_state(filter_id, project):
         raise FilterNotRegistered(filter_id)
 
     filter_state = ProjectOption.objects.get_value(
-        project=project, key=u"filters:{}".format(flt.spec.id)
+        project=project, key=u"filters:{}".format(flt.id)
```

Comment:
Commenting here because I can't below. `_filter_from_filter_id` needs the same patch, since it still accesses `flt.spec.id` but it should `flt.id`.

https://github.com/getsentry/sentry/blob/a974b513b3b3be119a594f5e71db0adfce1d81ef/src/sentry/message_filters.py#L114